relay-server/src/services/health_check.rs

```diff
@@ -214,16 +231,22 @@ impl Service for HealthCheckService {
             update_tx.send(StatusUpdate::new(Status::Unhealthy)).ok();
         });
 
-        while let Some(HealthCheck(message, sender)) = rx.recv().await {
-            let update = update_rx.borrow();
-
-            sender.send(if matches!(message, IsHealthy::Liveness) {
-                Status::Healthy
-            } else if update.instant.elapsed() >= status_timeout {
-                Status::Unhealthy
-            } else {
-                update.status
-            });
+        while let Some(message) = rx.recv().await {
+            match message {
+                HealthCheck::IsHealthy(message, sender) => {
+                    let update = update_rx.borrow();
+                    sender.send(if matches!(message, IsHealthy::Liveness) {
+                        Status::Healthy
```

Comment:
We considered failing the `Liveness` check once a service has crashed, but IIUC this would mean that Kubernetes would kill the process immediately. We want to keep the process alive so other services still have a chance to finish their work.

> If the liveness probe fails, the kubelet kills the container

https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#types-of-probe